core/unix/src/TUnixSystem.cxx

```diff
@@ -40,10 +40,15 @@
 #include <algorithm>
 #include <atomic>
 
+#ifdef __linux__
+#include <syscall.h>
+#endif
+
 //#define G__OLDEXPAND
 
 #include <unistd.h>
 #include <stdlib.h>
+#include <poll.h>
```

Comment:
Ah - TSystem::Select calculates its own bitmask instead of using `FD_SET` (see http://sourceforge.net/p/opalvoip/bugs/496/ for a decent explanation).  `FD_SET` is what causes the assert for large FDs.

Unfortunately, we can't use TSystem::Select because it internally uses `TFdSet`, which can lead to memory allocations (unsafe in the signal handler).

I guess we just have to use select on Windows and fail if fd > 1024.